### PR TITLE
Dev sir 1409 transfer details

### PR DESCRIPTION
**Pull request by aushakov**

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
       - name: Publish to Registry
         uses: swisschain/Publish-Docker-Github-Action@master
         with:
-          name: swisschains/sirius-vault-guardian
+          name: swisschains/sirius-vault-guardian:latest
```
Comment (KonstantinRyazantsev) [inline .github/workflows/gradle.yml:35]: why not `dev`?

File: src/main/java/io/swisschain/contracts/Asset.java [hunk 2]
```diff
@@ -36,4 +36,23 @@ public String getAddress() {
   public void setAddress(String address) {
     this.address = address;
   }
+
+  public boolean equal(Asset asset) {
```
Comment (dmitrykrasovskih) [inline src/main/java/io/swisschain/contracts/Asset.java:40]: Please use autogenerated equals and hashCode - they should be synchronized with each other.

File: src/main/java/io/swisschain/contracts/DestinationAddress.java [hunk 3]
```diff
@@ -59,4 +59,37 @@ public TagType getTagType() {
   public void setTagType(TagType tagType) {
     this.tagType = tagType;
   }
+
+  public boolean equal(DestinationAddress destinationAddress) {
```
Comment (dmitrykrasovskih) [inline src/main/java/io/swisschain/contracts/DestinationAddress.java:63]: Please use autogenerated equals and hashCode - they should be synchronized with each other.

File: src/main/java/io/swisschain/contracts/RequestContext.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+  public Instant getTimestamp() {
+    return timestamp;
+  }
+
+  public void setTimestamp(Instant timestamp) {
+    this.timestamp = timestamp;
+  }
+
+  public boolean equal(RequestContext requestContext) {
```
Comment (dmitrykrasovskih) [inline src/main/java/io/swisschain/contracts/RequestContext.java:52]: Please use autogenerated equals and hashCode - they should be synchronized with each other.

File: src/main/java/io/swisschain/contracts/SourceAddress.java [hunk 5]
```diff
@@ -36,4 +36,23 @@ public String getGroup() {
   public void setGroup(String group) {
     this.group = group;
   }
+
+  public boolean equal(SourceAddress sourceAddress) {
```
Comment (dmitrykrasovskih) [inline src/main/java/io/swisschain/contracts/SourceAddress.java:40]: Please use autogenerated equals and hashCode - they should be synchronized with each other.

File: src/main/java/io/swisschain/contracts/TransferContext.java [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+  public RequestContext getRequestContext() {
+    return requestContext;
+  }
+
+  /** Sets the client request details. */
+  public void setRequestContext(RequestContext requestContext) {
+    this.requestContext = requestContext;
+  }
+
+  public boolean equal(TransferContext transferContext) {
```
Comment (dmitrykrasovskih) [inline src/main/java/io/swisschain/contracts/TransferContext.java:128]: Please use autogenerated equals and hashCode - they should be synchronized with each other.

File: src/main/java/io/swisschain/contracts/TransferDetails.java [hunk 7]
```diff
@@ -22,78 +23,105 @@ public TransferDetails(
...
   }
 
-  public void setClientContext(ClientContext clientContext) {
-    this.clientContext = clientContext;
+  /** Sets the transfer detailed information and initial client request. */
+  public void setTransferContext(TransferContext transferContext) {
+    this.transferContext = transferContext;
+  }
+
+  public boolean equal(TransferDetails transferDetails){
```
Comment (dmitrykrasovskih) [inline src/main/java/io/swisschain/contracts/TransferDetails.java:117]: Please use autogenerated equals and hashCode - they should be synchronized with each other.

**Review by dmitrykrasovskih — CHANGES_REQUESTED** [review]

**Pull request merged**